azurerm/internal/services/lighthouse/tests/resource_arm_lighthouse_assignment_test.go

```diff
@@ -0,0 +1,216 @@
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMLighthouseAssignmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMLighthouseAssignment_basic(id, secondTenantID, principalID, data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLighthouseAssignmentExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "name"),
+				),
+			},
+			{
+				Config:      testAccAzureRMLighthouseAssignment_requiresImport(id, secondTenantID, principalID, data),
+				ExpectError: acceptance.RequiresImportError("azurerm_lighthouse_assignment"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMLighthouseAssignment_emptyID(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_lighthouse_assignment", "test")
+	secondTenantID := os.Getenv("ARM_TENANT_ID_ALT")
+	principalID := os.Getenv("ARM_PRINCIPAL_ID_ALT_TENANT")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMLighthouseAssignmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMLighthouseAssignment_emptyId(secondTenantID, principalID, data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLighthouseAssignmentExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "id"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "name"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMLighthouseAssignmentExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Lighthouse.AssignmentsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %q", resourceName)
+		}
+
+		scope := rs.Primary.Attributes["scope"]
+		lighthouseAssignmentName := rs.Primary.Attributes["name"]
+		expandLighthouseDefinition := true
+
+		resp, err := client.Get(ctx, scope, lighthouseAssignmentName, &expandLighthouseDefinition)
+
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Lighthouse Assignment %q (Scope: %q) does not exist", lighthouseAssignmentName, scope)
+			}
+			return fmt.Errorf("Bad: Get on LighthouseAssignmentsClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMLighthouseAssignmentDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Lighthouse.AssignmentsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_lighthouse_assignment" {
+			continue
+		}
+
+		scope := rs.Primary.Attributes["scope"]
+		lighthouseAssignmentName := rs.Primary.Attributes["name"]
+		expandLighthouseDefinition := true
```

Comment:
this can be `false` here to save retrieving all of the other information we don't need:

```suggestion
		expandLighthouseDefinition := false
```

Reply:
I removed that schema attribute already, and it's set to false in the API calls.  Since the tests won't work in our acceptance environment anyway I've not address the test code in the changes. We'll need to follow up when we're able to use binary testing.